cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -124,6 +125,15 @@ class LeadImpl extends ServerImpl with Lead with Logging {
         conf.set(key, v)
       })
 
+      /**
+       * If kryoserializer is being used then register snappyclasses for kryoserializer.
+       * The snappyclasses will not get registered if user is creating its own spark context.
+       * In that case user has to specify the snappy kryo registrator externally
+       */
+      if (conf.contains("spark.serializer") && conf.get("spark.serializer").equals("org.apache.spark.serializer.KryoSerializer")) {
+        conf.set("spark.kryo.registrator", getSnappyKryoRegistrarClass())
```

Comment:
Please check https://github.com/EsotericSoftware/kryo#registration also, where different registration methods are explained. It would be good if we can keep look and feel of Spark as it is, and can work through our inner details in non invasive way. If not anyway we have your code to fallback on.